### PR TITLE
Refactor/banner cards

### PR DESCRIPTION
**Pull request by dannyk3941**
#### What's this PR do?
- Refactor banner card pattern and add story controls
- Add story controls for General Card patterns

### CODE REVIEW

File: packages/vanilla/src/components/card/bannerCard.stories.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  parameters: {
+    layout: 'centered'
+  },
+  argTypes: {
+    title: {
+      name: 'Title',
+      description: 'Set the title in the banner area of the card',
+      control: 'text'
+    },
+    modifier: {
```
Comment (dgibson666) [inline packages/vanilla/src/components/card/bannerCard.stories.js:12]: Can we call this "Variant" instead of "Modifier"?

**Review by dgibson666 — COMMENTED** [review]
* Can we call the control "Variant" instead of "Modfier"? (commented in code as well)
* I think the story should use the info-circle icon. Using a letter-form icon next to text doesn't work well.
* Should the success/danger variants use different icons as well?
* The padding of the content region is doubling up with the margin from the paragraph. Perhaps a `:last-child` selector is needed to zero out the `padding-bottom` and `margin-bottom`.

Regarding the last item, another option would be to apply the light gray background to the parent wrapper and use margins on the content area. That way the margins will collapse without a `:last-child` selector.

File: packages/vanilla/src/components/card/bannerCard.stories.js (hunk 1 above)
Comment (dannyk3941) [reply packages/vanilla/src/components/card/bannerCard.stories.js]: Changed to 'variant'

**Review by dgibson666 — APPROVED** [review]
Looks good!

**Pull request merged**